src/ochre_simulation.py

```diff
@@ -0,0 +1,117 @@
+import os
+import xmltodict
+import pandas as pd
+import datetime as dt
+from datetime import date
+import time
+import shutil
+from ochre import Dwelling, Analysis, CreateFigures
+from ochre.utils import default_input_path  # for using sample files
+
+"""
+Code for running simulations on OCHRE.
+
+Author: Alex Lee (alexlee5124@gmail.com)
+Date: 4/19/2025
+
+"""
+
+def simulate(
+        building_id:int,
+        upgrade_id:int,
+        state:str,
+        year=2022,
+        version="resstock_amy2018_release_1.1",
+        start_time=dt.datetime(2007, 1, 1, 0, 0),
+        time_res=dt.timedelta(minutes=60),
+        duration=dt.timedelta(days=3),
+        ):
+    """
+    simulate: This function runs an OCHRE simulation for a building. It filters the building by the
+    building id, upgrade id, state, year, and version of the building XML file. You need to have already
+    downloaded the building XML file prior to running this function.
+
+    Input simulation parameters, such as duration, start date, simulation timestep size, etc.
+
+    Author: Alex Lee (alexlee5124@gmail.com)
+    Date: 3/29/2025
+    """
+
+    building_folder = f"bldg{building_id:07}-up{upgrade_id:02}"
+    input_file_path = os.path.join(os.path.dirname(os.path.dirname(os.getcwd())), f"data/building_energy_models/{state}_{year}_{version}/{building_folder}")
+
+    # Load XML and schedule file
+    xml_file = None
+    schedule_file = None
+    for file in os.listdir(input_file_path):
+        if file.endswith(".xml"):
+            xml_file = os.path.join(input_file_path, file)
+        elif file.endswith(".csv"):
+            schedule_file = os.path.join(input_file_path, file)
+
+    # Check that XML and schedule files exist. Import relevant if they do.
+    if xml_file:
+        # Extract weather station name if XML file exists
+        weatherStation = extract_weather_station(xml_file)
```

Comment:
Never use camelCase for anything in python, and variables are always lowercase_with_underscores. black or pep8 should alert you to style stuff list this.